PWGHF/TableProducer/trackToCollisionAssociator.cxx

```diff
@@ -0,0 +1,148 @@
+
+    // loop over collisions to find time-compatible tracks
+    float nSigmaForTimeCompat2 = nSigmaForTimeCompat * nSigmaForTimeCompat;
+    auto trackBegin = tracks.begin();
+    const auto bOffsetMax = 241;
+    for (const auto& collision : collisions) {
+      float collTime = collision.collisionTime();
+      float collTimeRes2 = collision.collisionTimeRes() * collision.collisionTimeRes();
+      uint64_t collBC = collision.bc().globalBC();
+      bool iteratorMoved = false;
+      for (auto track = trackBegin; track != tracks.end(); ++track) {
+        if (!track.has_collision()) {
+          continue;
+        }
+        int64_t bcOffset = (int64_t)track.collision().bc().globalBC() - (int64_t)collBC;
+        float trackTime{0.};
+        float trackTimeRes2{0.};
+        if (track.isPVContributor()) {
+          trackTime = track.collision().collisionTime(); // if PV contributor, we assume the time to be the one of the collision
+          trackTimeRes2 = 25.f;                          // 1 BC
+        } else {
+          trackTime = track.trackTime();
+          trackTimeRes2 = track.trackTimeRes() * track.trackTimeRes();
+        }
+        float deltaTime = trackTime - collTime + bcOffset * 25.f;
+        float sigmaTimeRes2 = collTimeRes2 + trackTimeRes2;
+        LOGP(debug, "collision time={}, collision time res={}, track time={}, track time res={}, bc collision={}, bc track={}, delta time={}", collTime, collision.collisionTimeRes(), track.trackTime(), track.trackTimeRes(), collBC, track.collision().bc().globalBC(), deltaTime);
+        if (!iteratorMoved && bcOffset > -bOffsetMax) {
+          trackBegin.setCursor(track.filteredIndex());
+          iteratorMoved = true;
+          LOGP(debug, "Moving iterator begin {}", track.globalIndex());
+        } else if (bcOffset > bOffsetMax) {
+          LOGP(debug, "Stopping iterator {}", track.globalIndex());
+          break;
+        }
+        if (deltaTime * deltaTime < nSigmaForTimeCompat2 * sigmaTimeRes2) {
+          auto collIdx = collision.globalIndex();
+          auto trackIdx = track.globalIndex();
+          LOGP(debug, "Filling track id {} for coll id {}", trackIdx, collIdx);
+          if (collsPerTrack[trackIdx] == nullptr) {
+            collsPerTrack[trackIdx] = new std::vector<int>;
+          }
+          collsPerTrack[trackIdx]->push_back(collIdx);
+          association(collIdx, trackIdx);
+        }
+      }
+    }
+
+    // create reverse index track to collisions
+    std::vector<int> empty{};
+    for (const auto& track : tracksUnfiltered) {
+      LOGP(debug, "Track id {}", track.globalIndex());
+      if (collsPerTrack[track.globalIndex()] == nullptr) {
+        reverseIndices(empty);
+      } else {
+        // for (const auto& collId : *collsPerTrack[track.globalIndex()]) {
+        //   LOGP(info, "  -> Coll id {}", collId);
+        // }
+        reverseIndices(*collsPerTrack[track.globalIndex()]);
+      }
```

Comment:
We can do the delete already here

Reply:
I removed the `delete` it given that I implemented Anton's suggestion with the `unique_ptr`